contrib/kafka/src/main/java/com/google/cloud/dataflow/contrib/kafka/coders/ConsumerRecordCoder.java

```diff
@@ -0,0 +1,52 @@
+package com.google.cloud.dataflow.contrib.kafka.coders;
+
+import com.google.cloud.dataflow.sdk.coders.CustomCoder;
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.codehaus.jackson.annotate.JsonCreator;
+
+/**
+ * A {@code ConsumerRecordCoder} encodes Kafka ConsumerRecord objects.
+ */
+public class ConsumerRecordCoder extends CustomCoder<ConsumerRecord<String, String>> {
+  private static final long serialVersionUID = 1L;
+
+
+  @JsonCreator
+  public static ConsumerRecordCoder of() {
+    return INSTANCE;
+  }
+
+  private static final ConsumerRecordCoder INSTANCE = new ConsumerRecordCoder();
+
+  private ConsumerRecordCoder() {}
+
+  @Override
+  public void encode(ConsumerRecord<String, String> value, OutputStream outStream, Context context)
+      throws IOException {
+    Gson gson = new Gson();
+    String json = gson.toJson(value);
+    outStream.write(json.getBytes(Charsets.UTF_8));
```

Comment:
might consider using versions that write to / read from streams directly ,e.g., https://sites.google.com/site/gson/streaming

(We actually use Jackson in Dataflow core SDK, not sure about pros/cons of Gson).